pkg/client/cli/cmd_gatherlogs.go

```diff
@@ -455,3 +463,43 @@ func anonymizeLog(stdout io.Writer, logFile string, anonymizer *anonymizer) erro
 
 	return nil
 }
+
+// getSignificantPodNames is a helper function that takes in a
+// pod's name and returns the significant subnames that we want
+// to anonymize.  It currently works for pods owned by StatefulSets,
+// ReplicaSets, and Deployments.
+func getSignificantPodNames(podName string) []string {
+	subNames := make([]string, 2)
+	nameComponents := strings.Split(podName, "-")
+	significantComponent := false
+	for i, component := range nameComponents {
+		// Indicates we found a Pod that comes from either a
+		// ReplicaSet and/or a Deployment
+		if _, err := hex.DecodeString(component); err == nil {
+			significantComponent = true
+		}
+		// Indicates we found a Pod that comes from a StatefulSet
+		if _, err := strconv.Atoi(component); err == nil {
+			significantComponent = true
+		}
+
+		if significantComponent {
+			nameComponents = nameComponents[:i+1]
+			break
+		}
+	}
```

Comment:
hmm im not sure im sold -- what if i have a deployment called `hello-0`? i almost wanna say we should loop in reverse; what about something like this:

```go
lastSignificant := false
subNames := []string{}
for i := len(nameComponents)-1; i >= 0; i-- {
                component := nameComponents[i]
                significant := false
                // Indicates we found a Pod that comes from either a
                // ReplicaSet and/or a Deployment
                if _, err := hex.DecodeString(component); err == nil {
                    significant = true
                }
                // Indicates we found a Pod that comes from a StatefulSet
                if _, err := strconv.Atoi(component); err == nil && !lastSignificant {
                    significant = true
                }
                
                if !significant && lastSignificant {
                     // my-sts or my-deploy
                     subNames = append(subNames, strings.Join(nameComponents[:i], "-"))
                     break
                } else if significant {
                     // my-sts-0 or my-deploy-asdf-asdf or my-deploy-asdf
                     subNames = append(subNames, strings.Join(nameComponents[:i], "-"))
                     lastSignificant = true
                }
}
```

although now that i look at that i think it won't work for a deployment called `a123f` 🤔 it might be easier to just do this with regexes 😅 should be easy to make one for statefulsets, one for deployments, and one for replicasets and see which is matched

Reply:
yeah there wasn't really any way that I was super happy with. I will try to think of a regexes.